metrics/stackdriver_exporter.go

```diff
@@ -63,10 +65,18 @@ func newOpencensusSDExporter(o stackdriver.Options) (view.Exporter, error) {
 // TODO should be properly refactored to be able to inject the getMonitoredResourceFunc function.
 // 	See https://github.com/knative/pkg/issues/608
 func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	gm := gcpMetadataFunc()
+	gm := getGCPMetadata(config)
 	mtf := getMetricTypeFunc(config.stackdriverMetricTypePrefix, config.stackdriverCustomMetricTypePrefix)
-	e, err := newStackdriverExporterFunc(stackdriver.Options{
-		ProjectID:               config.stackdriverProjectID,
+	co, err := getStackdriverExporterClientOptions(&config.stackdriverClientConfig)
+	if err != nil {
+		logger.Warnw("Issue configuring Stackdriver exporter client options, no additional client options will be used: ", zap.Error(err))
```

Comment:
This warning reads to me like a minor issue and I'm not sure everyone will know what this means exactly. If I'm reading the code correctly, it seems that this failure case is more about accessing the configured secret, and not about interacting with the stackdriver client. Thoughts on something like "Failed to read configured Stackdriver secret"? (which might make more sense down in `getStackdriverExporterClientOptions()`)

Reply:
I added some formatting from the error that gets emitted from getStackdriverSecret to print more info. It will get printed here as part of the `err` field, and should be actionable by the user to fix.

I think it's important to include a bit about the Stackdriver exporter client options so users understand the end result of the Secret failing.

Reply:
Looks great, thanks!